File: java/src/main/java/rabbitmqinaction/sourcecode/GenericConfiguration.java
package rabbitmqinaction.sourcecode;

import java.util.HashMap;
import java.util.Map;

public class GenericConfiguration {

    public static final boolean ACTIVE = false;
    public static final boolean PASSIVE = false;
    public static final boolean DURABLE = true;
    public static final boolean NON_DURABLE = false;
    public static final boolean AUTO_DELETE = true;
    public static final boolean NON_AUTO_DELETE = false;
    public static final String DIRECT_EXCHANGE_TYPE = "direct";
    public static final String HOST = "localhost";
    public static final String PLAIN_CONTENT_TYPE = "text/plain";
    public static final Map<String, Object> EMPTY_MAP = new HashMap<String, Object>();

}

File: scala/src/main/scala/rabbitmqinaction/sourcecode/GenericConfiguration.scala
/**
 * RabbitMQ in Action - Generic constants
 *
 * @author Simon Fraser, Siniatech Ltd 
 */
package rabbitmqinaction.sourcecode

object GenericConfiguration {
  val Active = false
  val Passive = false
  val Durable = true
  val NonDurable = false
  val AutoDelete = true
  val NonAutoDelete = false
  val DirectExchangeType = "direct"
  val PlainContentType = "text/plain"
  val EmptyJavaMap = new java.util.HashMap[String, Object]
  def host = "localhost"
}